internal/config/instance.go

```diff
@@ -26,6 +26,7 @@ type Instance struct {
 	thread      *singlethread.Thread
 	closeThread bool
 	db          *sql.DB
+	closed      bool
```

Comment:
This should probably be guarded by a mutex.  Or maybe you can use a `sync.Once` instance in the `Close()` function to ensure that the thread is closed only once.